sql/lesson2/tutorial.md

```diff
@@ -0,0 +1,616 @@
+
+Normally, when you write a query, you'll want your results to be shown in a specific order. Doing so is very simple knowing the operator **ORDER BY**.
+This operator is placed at the end of the **SELECT** statement. It's basic syntax is the following:
+
+```SQL
+SELECT {columns}
+  FROM {table name}
+ WHERE {conditions}
+ ORDER BY {expression} {direction};
+```
+Where:
+
+* **SELECT**, **FROM**, and **WHERE** are the same as in previous lesson,
+* **{expression}** can be a _column_ name, the _column_ position, a function, or an arithmetic expression,
+* **{direction}** is either ASC or DESC, if omitted ASC is assumed.
+
+Using the same database as in Lesson 1 (https://sqliteonline.com/#fiddle-5be8825d749e251ljod9s39t), run the following query.
+```SQL
+SELECT name, age
+  FROM people
+ ORDER BY age DESC;
+```
+
+Is the result sorted by descending age? What happen in ages that appear more than once? Are the names sorted?
+
+You can add more than one order by pairs of expression and direction:
+
+```SQL
+SELECT name, age
+  FROM people
+ ORDER BY age DESC, name ASC;
+```
+
+With the SQL above, in case of equal ages, then records will be sorted by name. Note that each sort has a different direction.
+
+The following query is equivalent to the previous one, but instead of using column names, column positions are used.
+```SQL
+SELECT name, age
+  FROM people
+ ORDER BY 2 DESC, 1 ASC;
+```
+
+This next query is an example of function. The function **LENGTH** in SQLite returns the length of the string passed as parameter (between the parentheses):
+```SQL
+SELECT name
+  FROM people
+ ORDER BY LENGTH(name) DESC;
+```
+
+Lets now use an arithmetic expression as a last example of sorting:
+```SQL
+SELECT name
+  FROM people
+ ORDER BY (age + 2 * number_of_children) DESC;
+```
+
+TIP: You can also add expressions and functions in the columns to show in the results: Try this out:
+```SQL
+SELECT name, (age + 2 * number_of_children)
+  FROM people
+ ORDER BY (age + 2 * number_of_children) DESC;
+```
+
+### Limiting the number of results and paginating
+
+When you are building queries that will fill reports, when you are showing the results in pages, or simply when the result is too long, you may want to limit the number of rows you show. To do so, you can use the operator **LIMIT** as follows
+
+```SQL
+SELECT name, age
+  FROM people
+ ORDER BY age DESC
+ LIMIT 5;
+```
+
+Contrary to the query used before without LIMIT that returned all the rows in the table, this one only returns the first 5.
+
+Although it is not strictly necessary to have an ORDER BY clause when you LIMIT the results, it is highly recommended. Doing so, you can be certain that you will always get the same results when running the same query multiple times.
+
+If additionally you are paginating the result, you need to use LIMIT together with **OFFSET**. This operator can we understood as _Skip the first N records, and give me the ones after_. For example, if you run
```

Comment:
Maybe you should start this para more like `If you wanted to view the next _x_ results, you can specify an ***OFFSET***`? Also, I have never used OFFSET in my entire SQL career, I always use the shortened `LIMIT <count>,<offset>` format. Happy to conceded that's a MySQL thing, but just wondered if it was also worth mentioning here? Or is this a coaches easter egg? ;)

Reply:
I learnt something :) 
I'm adding the following:

There is a short form when using LIMIT and OFFSET together. The following two queries produce the same results:

```SQL
SELECT name, age
  FROM people
 ORDER BY age DESC
 LIMIT 5
OFFSET 3;
```
```SQL
SELECT name, age
  FROM people
 ORDER BY age DESC
 LIMIT 3,5;
```

When using this short form, the first number corresponds to the _offset_ and the second one to the _number of records returned_.